sdk/storage/azure-storage-blob/v10_v12.md

```diff
@@ -0,0 +1,437 @@
+```java
+// Here is how the blob object take in the sas token generated in above section.
+BlobURLParts parts = URLParser.parse(cu.createBlobURL(blobName).toURL())
+parts.withSasQueryParameters(sasToken).withScheme("https")
+BlockBlobURL bu = new BlockBlobURL(parts.toURL(), StorageURL.createPipeline(
+    new AnonymousCredentials(), new PipelineOptions()))
+```
+
+In v12, SAS tokens can be managed just like any other credential.
+```java
+// Initialize the credentials
+SASTokenCredential credential = SASTokenCredential.fromSASTokenString("${sasToken}");
+// How service object take in the credentials.
+BlobServiceClient client = new BlobServiceClientBuilder()
+    .endpoint("https://${accountName}.blob.core.windows.net")
+    .credential(credential)
+    .buildBlobServiceClient();
+```
+
+However, if you already have the full URL with SAS token attached, you can use that too:
+```java
+BlobServiceClient client = new BlobServiceClientBuilder()
+    .endpoint("https://${accountName}.blob.core.windows.net/?${sasToken}")
+    .buildBlobServiceClient();
+```
+
+## Provide side by side sync-over-async APIs
+Each type (Container, Blob, etc.) has both a sync and async client. It is easy and intuitive to use whichever one is preferable in a given context without having to bother about clutter from the other.
+Suppose the Container Client been initialized as `${containerClient}`. Please refer the [Build Client](#build-client) section.
+
+* Example of using Async API in V12. 
+```java
+Mono<BlobProperties> response = containerClient.getProperties();
+```
+
+* Example of using Sync API in V12
+```java
+BlobProperties blobProperties = containerClient.getProperties();
+```
+
+## Async APIs Comparison
+In v10/v11, every RESTful API returned a reactive response of type `Single`
+(see ReactiveX documentation for more details), which emits one and only one result.
+These returned object representations of the single HTTP response that came back.
+
+Single result API in v10/v11:
+```java
+serviceURL.getProperties(Context.NONE).subscribe(
+    // print the selected service property on success
+    response -> System.out.println(response.body().defaultServiceVersion()),
+    // handle error
+    error -> System.out.println(error.getMessage()));
+```
+
+Listing API in v10/v11:
+```java
+serviceURL.listContainersSegment(null, null, Context.NONE).subscribe(
+    // print the list; grab continuation token for making another call to get next list segment
+    response -> {
+        System.out.println("Page token to fetch more containers:" + response.body().nextMarker());
+        response.body().containerItems().stream().forEach(
+            containerItem -> System.out.println("Container name: " + containerItem.name()));
+    },
+    // handle error
+    error -> System.out.println(error.getMessage()));
+```
+
+In v12, Reactor provides an equivalent class `Mono` (0-1 response) and provide our own extension of Reactor's `Flux` (0-many responses) for listing operations.
```

Comment:
```suggestion
In v12, Reactor provides an equivalent class `Mono` (0-1 response). There is also the `Flux` class for zero-to-many responses, although Azure client libraries use a subclass known at `PagedFlux` to also provide pagination and response information.
```

Reply:
I wouldn't use "althought" because that makes it sound like we never return a normal Flux, which we do for downloads.